Dynavity/Dynavity/model/AnnotationPalette.swift

```diff
@@ -1,54 +1,61 @@
 import PencilKit
 
 struct AnnotationPalette {
+    enum SelectedTool {
+        case pen
+        case marker
+        case eraser
+        case lasso
+    }
+
     static let annotationWidths: [CGFloat] = [3, 6, 9, 12]
     static let annotationColors = [UIColor.blue, UIColor.red, UIColor.green, UIColor.yellow]
-    static let defaultPenTool = PKInkingTool(.pen, color: .blue, width: annotationWidths[2])
-    static let defaultMarkerTool = PKInkingTool(.marker, color: .yellow, width: annotationWidths[2])
-
-    var selectedTool: PKTool
+    private var penTool = PKInkingTool(.pen, color: .blue, width: annotationWidths[2])
+    private var markerTool = PKInkingTool(.marker, color: .yellow, width: annotationWidths[2])
+    private var selectedTool: SelectedTool
 
     init() {
-        selectedTool = AnnotationPalette.defaultPenTool
+        selectedTool = .pen
     }
 
-    mutating func switchTool(_ newTool: PKTool) {
+    mutating func switchTool(_ newTool: SelectedTool) {
         selectedTool = newTool
     }
 
-    mutating func switchAnnotationWidth(_ newWidth: CGFloat) {
-        guard let inkingTool = selectedTool as? PKInkingTool else {
-            return
-        }
-
-        // Only pen and and marker (highlighter) inking tools used
-        switch inkingTool.inkType {
+    func getSelectedTool() -> PKTool {
+        switch selectedTool {
         case .pen:
-            selectedTool = PKInkingTool(.pen, color: inkingTool.color, width: newWidth)
+            return penTool
         case .marker:
-            selectedTool = PKInkingTool(.marker, color: inkingTool.color, width: newWidth)
-        case .pencil:
-            fatalError("PKInkingTool should not be pencil")
-        @unknown default:
-            fatalError("PKInkingTool is not a pen or marker")
+            return markerTool
+        case .eraser:
+            return PKEraserTool(.vector)
+        case .lasso:
+            return PKLassoTool()
         }
     }
 
-    mutating func switchAnnotationColor(_ newColor: UIColor) {
-        guard let inkingTool = selectedTool as? PKInkingTool else {
-            return
+    mutating func setAnnotationWidth(_ newWidth: CGFloat) {
+        switch selectedTool {
+        case .pen:
+            penTool.width = newWidth
+        case .marker:
+            markerTool.width = newWidth
+        default:
+            // Do nothing.
```

Comment:
I would prefer to `fatalError` here instead since it helps us to verify that end-users are not allowed to change the width when a non-pen / marker tool is selected. And if for some reason end-users are able to do so (e.g. a lapse in our UI), we would be immediately notified rather than things failing silently.

Likewise for setting of annotation colour

Reply:
That makes sense, thanks!